store/sqlstore/post_store.go

```diff
@@ -254,18 +256,24 @@ func (s *SqlPostStore) Update(newPost *model.Post, oldPost *model.Post) (*model.
 	}
 
 	if _, err := s.GetMaster().Update(newPost); err != nil {
-		return nil, model.NewAppError("SqlPostStore.Update", "store.sql_post.update.app_error", nil, "id="+newPost.Id+", "+err.Error(), http.StatusInternalServerError)
+		return nil, errors.Wrapf(err, "failed to update Post with id=%s", newPost.Id)
 	}
 
 	time := model.GetMillis()
-	s.GetMaster().Exec("UPDATE Channels SET LastPostAt = :LastPostAt  WHERE Id = :ChannelId AND LastPostAt < :LastPostAt", map[string]interface{}{"LastPostAt": time, "ChannelId": newPost.ChannelId})
+	if _, nErr := s.GetMaster().Exec("UPDATE Channels SET LastPostAt = :LastPostAt  WHERE Id = :ChannelId AND LastPostAt < :LastPostAt", map[string]interface{}{"LastPostAt": time, "ChannelId": newPost.ChannelId}); nErr != nil {
```

Comment:
You are changing the behavior here, this is intentionally fire and forget, because if this fails, we don't care, will be eventually consistent in the future with the next post, is something that is not going to happen often, and is something that we prefer to not break the new post creation because of this. Maybe we can add logs here instead of returning the error.

Reply:
Yes I know I'm changing the behaviour here and I understand your point. Agree with you that maybe it is not very important to update the Channel for this case but any error received while doing an sql operation should be handled no matter it is important or no for our business rules. I would understand that this for cases when for example the error could be `sql.ErrNoRows` because it is expected in some cases not finding some values for bad requests.

For this case I think we should handle the error not because it is important to update the channel itself, but simply an error here should not happen and if that is ocurring then ignore or not handle meas we are ignoring a possible future error, perhaps something like logging it is less drastic.

Reply:
Finally I'm going to leave the code as is. We can get into this in another PR. Thanks @jespino